### ISSUE TITLE
Problem with SEF

### ISSUE THREAD

**Issue by TrueHatori** [e107inc/e107#3818]
I have a problem with the search engine friendly URLs. The news works wonderfully. Look here:
![image](https://user-images.githubusercontent.com/17859550/58702853-3ac0c600-83a7-11e9-8a80-c87881c72248.png)

However, it does not work on the pages.
![image](https://user-images.githubusercontent.com/17859550/58702940-722f7280-83a7-11e9-9e7f-106918d27bea.png)

If I call a page from the admin area, right-click on the ID, then it works.
![image](https://user-images.githubusercontent.com/17859550/58703031-aefb6980-83a7-11e9-9e1b-52c67af5bb92.png)
![image](https://user-images.githubusercontent.com/17859550/58703061-c76b8400-83a7-11e9-97bf-5b7767f112cf.png)

This is nothing new in v2.2.1, that was already the case in v2.2.0.

![image](https://user-images.githubusercontent.com/17859550/58703141-fbdf4000-83a7-11e9-8bd9-fc1f12b2ec86.png)

**Comment by Jimmi08** [comment]
Hm, you have not SEF URLs in your site links.  I am not aware this worked itself (automatically) anytime.   Just find SEF-URL (from admin) and then replace it in site links.  And welcome to double content issue with google. 
PS. just to be sure - post here your actual site links setting for page 9 (http://ninja4ever.de/seite/dojo-infos/)

**Comment by TrueHatori** [comment]
I'm not sure, if I understand, what you mean :(
I think, you mean that:
![image](https://user-images.githubusercontent.com/17859550/58760965-fcf6a580-853e-11e9-8734-50a35b1d21ce.png)

But URL is with number:
![image](https://user-images.githubusercontent.com/17859550/58760995-30d1cb00-853f-11e9-8228-aad0bff818f6.png)

**Comment by Jimmi08** [comment]
No, your sitelinks  for this exact page.  Maybe you have there "hardcoded" url. Example: 

![image](https://user-images.githubusercontent.com/5429548/58761045-dedd7500-853f-11e9-855a-43d3ec28be18.png)

**Comment by TrueHatori** [comment]
Ok, yes, that's how it looks to me. Then the numbers stay, I have no desire or time to do it all by hand. If anyone wants to find our site, find us without SEFs. Thanks.

**Comment by Moc** [comment]
@CaMer0n Would be interesting to render the SEF URL if enabled, and filled in, right?

**Comment by CaMer0n** [comment]
@Moc I'm not sure I'm following the issue here. 
The value stored in sitelinks is added automatically during installation of the plugin. It is the 'key' that should be used when generating the SEFURL. ie. the one of the keys defined in e_url.php. 

Example: 

### gallery/plugin.xml

![image](https://user-images.githubusercontent.com/2825679/58912492-ff880500-86ce-11e9-908c-f6b98df85e34.png)

### gallery/e_url.php:
```
class gallery_url // plugin-folder + '_url'
{
	function config()
	{
		$config = array();

		$config['index'] = array(
		//	'regex'			=> '^gallery/?$', 						// matched against url, and if true, redirected to 'redirect' below.
			'alias'         => 'gallery',
			'sef'			=> '{alias}', 							// used by e107::url(); to create a url from the db table.
			'redirect'		=> '{e_PLUGIN}gallery/gallery.php', 		// file-path of what to load when the regex returns true.
			'legacy'        => '{e_PLUGIN}gallery/gallery.php',
		);


		return $config;
	}



}
```

**Comment by Jimmi08** [comment]
@CaMer0n 
I understood this like:
- you have page f.e. with ID 9
- you added it to sitelinks (no idea if manually, by the installation or by generating from pages)
- then you change URL configuration and users are waiting that it will change itself to SEF version.

I tried to explain that it depends on how it was added to sitelinks.

**Comment by CaMer0n** [comment]
@Jimmi08 I understand.  I guess the only work-around would be to store the SEF and non-SEF urls in the sitelinks table when it is generated. Of course, it would need to be manually edited if the page SEF url is ever modified.

**Comment by Jimmi08** [comment]
@CaMer0n There is already solution with sitelinks function.  All pages or All pages from the chapter. I suppose it uses inside e107::url(), so it's always correct. 

And with generating sitelinks directly from pages... it's not so hard to change this manually.  People just don't know about those possibilities.

**Comment by CaMer0n** [comment]
@Jimmi08 Yes, the sitelinks function is a good solution for sub-links.
